.pipelines/templates/stages/testing_rollback/testing-template.yml

```diff
@@ -0,0 +1,165 @@
+    values:
+      - qemu-grub
+      - qemu
+      - uki
+
+  - name: pool
+    displayName: Agent pool
+    type: string
+    default: "trident-ubuntu-1es-pool-eastus2"
+  
+  - name: skipExtensionTesting
+    displayName: "Skip extension testing"
+    type: string
+    default: "false"
+
+  - name: skipManualRollbackTesting
+    displayName: "Skip manual rollback testing"
+    type: string
+    default: "true"
+
+  - name: skipRuntimeUpdateTesting
+    displayName: "Skip runtime update testing"
+    type: string
+    default: "true"
+
+jobs:
+  - job: RollbackTesting_${{ replace(parameters.flavor, '-', '_') }}
+    displayName: Rollback Testing - ${{ parameters.flavor }}
+    timeoutInMinutes: ${{ parameters.updateCheckTimeoutInMinutes }}
+    pool:
+      type: linux
+      name: ${{ parameters.pool }}
+      hostArchitecture: amd64
+
+    variables:
+      tridentSourceDirectory: $(Build.SourcesDirectory)
+      ob_outputDirectory: $(tridentSourceDirectory)/deployment_logs_${{ parameters.flavor }}
+      ob_artifactBaseName: "rollback-testing-${{ parameters.flavor }}"
+      TEST_RESOURCE_GROUP: trident-vm-servicing-validation-$(Build.BuildId)
+
+    steps:
+      - task: DownloadPipelineArtifact@2
+        inputs:
+          buildType: current
+          artifactName: image-${{ parameters.flavor }}-base
+          targetPath: "$(Build.ArtifactStagingDirectory)/"
+        displayName: Download Base Image
+
+      - task: DownloadPipelineArtifact@2
+        inputs:
+          buildType: current
+          artifactName: image-${{ parameters.flavor }}-update-a
+          targetPath: "$(Build.ArtifactStagingDirectory)"
+        displayName: Download Update Image
+
+      - task: DownloadPipelineArtifact@2
+        inputs:
+          buildType: current
+          artifactName: ssh-keys
+          targetPath: "$(Build.ArtifactStagingDirectory)/ssh"
+        displayName: Download SSH Keys
+
+      - task: DownloadPipelineArtifact@2
+        displayName: "Download go-tools"
+        inputs:
+          buildType: current
+          artifactName: "go-tools"
+          patterns: |
+            netlisten
+            storm-trident
+          targetPath: "$(tridentSourceDirectory)/bin"
+
+      - bash: |
+          set -eux
+          chmod +x $(tridentSourceDirectory)/bin/netlisten
+          chmod +x $(tridentSourceDirectory)/bin/storm-trident
+          cp $(Build.ArtifactStagingDirectory)/ssh/id_rsa* ~/.ssh/
+          chmod -R 700 ~/.ssh/
+          mkdir -p $(ob_outputDirectory)
+          ls -l ~/.ssh/
+        displayName: Set up SSH Keys and setup output directory
+
+      - bash: |
+          set -eux
+          ./bin/storm-trident script build-extension-images --build-sysexts --num-clones 3
+          mv ./test-sysext-*.raw $(Build.ArtifactStagingDirectory)/
+
+          find $(Build.ArtifactStagingDirectory)
+        displayName: "Create extensions for test"
+
+      - bash: |
+          set -eux
+          
+          STORM_DYNAMIC_FLAGS=""
```

Comment:
i'd argue that all this behavior should be internal to storm and we should minimize the amount of logic we encode in bash/yaml